### DISCUSSION TITLE
Content Panel not loading in some cases

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#500] [Q&A] [answered]
Does anyone else see an issue where if you start routing to a new form and you try start routing to a second form before the first has loaded, nothing actually loads to the content panel?

This is particularly common if the first form has a long `__init__()` like if it has a server call in it?

If it does error, navigating back and then forward on the browser or routing seems to render the form correctly.

I can build a minimal example if required or screen share to show this in reality.

**Comment by s-cork** [comment, marked as answer]
Yes minimal example please (always).

**Reply by rhurlbatt** [reply]
Sorry should have done this by default.. noted thank you.

Ok so here is an example and clone
[Example Link](https://embarrassed-jumpy-electric-ray.anvil.app)
[Clone Link](https://anvil.works/build#clone:IPTBHBTBL3NCEHAN=6OFKKFSYVFX2LTD2HTCTBGAZ)
[Video of failing navigation](https://app.screencastify.com/v3/watch/XfmwzjXRsv2yoJ1ginQh)

It seems that this only happens when there is a server call on both forms. Print statements happen from both server calls but no form is loaded into the content panel.
Pressing back and forward renders the form correctly.

If you comment out the server call on the second form when you click it it loads instantly even while the first form is still loading.

Thanks for having a look Stu.

**Reply by rhurlbatt** [reply]
This is working beautifully now. Thank you very much Stu.
